### PR TITLE
Basements update

### PR DESCRIPTION
**Pull request by chaisawlajatang**
#### Summary

```SUMMARY: Content "Updates house basements'"```

#### Purpose of change

Weed basement and chem basement are in fact huge cold empty rooms. Time to give them some love.

#### Describe the solution

I changed them to more interesting (I believe) layout. Chem basement now has two variations - "home workshop" and a slightly more rare "home laboratory". I included another basement type - a tight messed room filled with random old junk. Also I adjusted weights, so there will be more variety underground (I barely saw residential basements before this).

#### Describe alternatives you've considered

Leave everything as is.
Update everything in one go.
Leave old empty rooms along with the new ones.

#### Testing

Created a half-dozen of new worlds, jumped around, checked out new basements and ensured they work as designed. Everything looks fine.

#### Additional context

I plan to maybe remove hardcoded basements (basement_junk, namely) and replace them with jsonized ones (jsonizing spider basements, of course). Also I plan to update other basements, add more variety to them, add special rare rooms and make more varieties of the messed basement.

Chem (lab)
chem
Chem (workshop)
chem1
Messed
mess
Weed
weed

### CODE REVIEW

File: data/json/mapgen/basement/basement_messed.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          { "item": "magazines", "chance": 60, "repeat": [ 0, 3 ] },
+          { "item": "homebooks", "chance": 50 },
+          { "item": "exotic_books", "chance": 5 }
+        ],
+        "b": [ { "item": "home_hw", "chance": 60 }, { "item": "mechanics", "chance": 30 } ]
+      },
+      "place_monsters": [ { "monster": "GROUP_ZOMBIE", "x": [ 9, 16 ], "y": [ 12, 16 ] } ]
+    }
+  }
+]
```
Comment (anothersimulacrum) [inline data/json/mapgen/basement/basement_messed.json]:
```suggestion
]

```

File: data/json/mapgen/basement/basement_chem.json [hunk 2]
```diff
@@ -3,43 +3,194 @@
...
+        "i": "f_sink",
+        "?": "f_sofa"
+      },
+      "items": {
+        "U": [ { "item": "cleaning", "chance": 30 }, { "item": "home_hw", "chance": 50 } ],
+        "c": [ { "item": "home_hw", "chance": 70 } ],
+        "R": [ { "item": "supplies_hardware", "chance": 30 } ],
+        "X": [
+          { "item": "electronics", "chance": 30 },
+          { "item": "supplies_electronics", "chance": 30 },
```
Comment (snipercup) [inline data/json/mapgen/basement/basement_chem.json]: Did you think about alternatives for these? I can't really explain why they would show up in the chem lab?

Comment (chaisawlajatang) [reply data/json/mapgen/basement/basement_chem.json]: It's not a pure chem lab (there's a meth basement for this), it's more like a "generic home laboratory".

**Comment by chaisawlajatang** [comment]
> If you're serious about working on the basements you might want to look into Specific Use Storage #29706 and nested chunks.

Yeah, I plan to do things with nested chunks (I thought about another place for them but basements would be okay too), and thanks for the link!

File: data/json/mapgen/basement/basement_chem.json (hunk 2 above)
Comment (snipercup) [reply data/json/mapgen/basement/basement_chem.json]: Ah okay I can see where you're coming from.

**Comment by curstwist** [comment]
really great, I like the varied sizes too.

**Comment by chaisawlajatang** [comment]
Hmmm, I wonder why Travis is complaining.

**Comment by Night-Pryanik** [comment]
It's complaining on every PR last days, don't pay attention to it.

**Comment by ZhilkinSerg** [comment]
> 
> 
> Hmmm, I wonder why Travis is complaining.

There is Clang check failing (will be addressed in #34823) and Android build timeout (was addressed in #34806).

**Pull request merged**